test/functional/terminal/ex_terminal_spec.lua

```diff
@@ -42,35 +42,53 @@ describe(':terminal', function()
 
 end)
 
-describe(':terminal (with fake shell)', function()
+describe(':terminal (with fake shell with arguments)', function()
   local screen
 
+  local function terminal_with_fake_shell(cmd)
+    execute("terminal "..(cmd and cmd or ""))
+  end
+
   before_each(function()
     clear()
     screen = Screen.new(50, 4)
     screen:attach({rgb=false})
     -- shell-test.c is a fake shell that prints its arguments and exits.
-    nvim('set_option', 'shell', nvim_dir..'/shell-test')
+    nvim('set_option', 'shell', nvim_dir..'/shell-test -t jeff')
```

Comment:
Can just do this at the start of a new `it()` case, instead of duplicating the full `describe()` block.